### ISSUE TITLE
Receiver fails to close channel when basicConsume is cancelled

### ISSUE THREAD

**Issue by fridgebuzz** [reactor/reactor-rabbitmq#85]
If a queue is deleted, the Receiver consumeXXX methods will attempt basicCancel(consumerTag) (for example, see Receiver line 179 for the consumeManualAck case). However this throws an exception:

> Error while closing channel: Unknown consumerTag

This causes the channel.close() (line 180) to be skipped and the catch clause just logs. This results in the channel remaining open forever.

One possible solution is to close the channel even if an exception is thrown here. Note the comments in the catch clause:

 ```
    // Not sure what to do, not much we can do,
    // logging should be enough.
    // Maybe one good reason to introduce an exception handler to choose more easily.
    LOGGER.warn("Error while closing channel: " + e.getMessage());
```

Exactly why the basicCancel fails in this case, I did not determine, so if there's a better solution that's great. A finally clause that closes the channel and swallows any exceptions would get the job done, though.

**Comment by fridgebuzz** [comment]
Actually, it looks like this may be fixed in 1.2.0 (released the same day as my report):

https://github.com/reactor/reactor-rabbitmq/pull/60

I think this is closable.

**Comment by acogoluegnes** [comment]
Yes, sounds like #60, which was fixed in 1.2.0. Don't hesitate to report back if you stumble on anything related to this issue.
